runtime-modules/council/src/lib.rs

```diff
@@ -0,0 +1,711 @@
+        Balance = Balance::<T>,
+        <T as system::Trait>::AccountId,
+        <T as Trait>::CouncilUserId,
+    {
+        /// New council was elected
+        AnnouncingPeriodStarted(),
+
+        /// Announcing period can't finish because of insufficient candidtate count
+        NotEnoughCandidates(),
+
+        /// Candidates are announced and voting starts
+        VotingPeriodStarted(u64),
+
+        /// New candidate announced
+        NewCandidate(Candidate<AccountId, Balance>),
+
+        /// New council was elected and appointed
+        NewCouncilElected(Vec<CouncilMember<AccountId, CouncilUserId, Balance>>),
+
+        /// New council was elected and appointed
+        NewCouncilNotElected(),
+    }
+}
+
+decl_error! {
+    /// Council errors
+    pub enum Error for Module<T: Trait> {
+        /// Origin is invalid
+        BadOrigin,
+
+        /// User tried to candidate outside of the announcement period
+        CantCandidateNow,
+
+        /// User tried to candidate outside of the announcement period
+        CantReleaseStakeNow,
+
+        /// Candidate haven't provide sufficient stake
+        CandidacyStakeTooLow,
+
+        /// Council member and candidates can't withdraw stake
+        StakeStillNeeded,
+
+        /// Candidate can't vote for himself
+        CantVoteForYourself,
+
+        /// Invalid runtime implementation broke the council. This error shouldn't happen
+        /// and in case of bad implementation should be discovered in the first block when referendum start will fail.
+        InvalidRuntimeImplementation,
+
+        /// Invalid membership
+        CouncilUserIdNotMatchAccount,
+    }
+}
+
+impl<T: Trait, RT: ReferendumTrait<I>, I: ReferendumInstanceGeneric> From<referendum::Error<RT, I>>
+    for Error<T>
+{
+    fn from(_other: referendum::Error<RT, I>) -> Error<T> {
+        //panic!(format!("{:?}", other)); // temporary debug
+        Error::<T>::BadOrigin // TODO: find way to select proper error
+    }
+}
+
+impl<T: Trait> PartialEq for Error<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_u8() == other.as_u8()
+    }
+}
+
+impl<T: Trait> From<BadOrigin> for Error<T> {
+    fn from(_error: BadOrigin) -> Self {
+        Error::<T>::BadOrigin
+    }
+}
+
+/////////////////// Module definition and implementation ///////////////////////
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        /// Predefined errors
+        type Error = Error<T>;
+
+        /// Setup events
+        fn deposit_event() = default;
+
+        /////////////////// Lifetime ///////////////////////////////////////////
+
+        // No origin so this is a priviledged call
+        fn on_finalize(now: T::BlockNumber) {
+            Self::try_progress_stage(now);
+        }
+
+        /// Subscribe candidate
+        #[weight = 10_000_000]
+        pub fn announce_candidacy(origin, council_user_id: T::CouncilUserId, stake: Balance<T>) -> Result<(), Error<T>> {
```

Comment:
Where in this code path is it ensured that the user is not already a candidate? It appears to just overwrite the candidate, but what does that doe with

Reply:
Repeated candidacy is prevented now.